src/xhr.ts

```diff
@@ -41,6 +41,39 @@ export class JQueryRequestor extends Requestor {
   }
 }
 
+/**
+ * Uses fetch API to make Ajax requests
+ */
+export class FetchRequestor extends Requestor {
+  xhr<T>(settings: JQueryAjaxSettings): Promise<T> {
+    const url: string = settings.url || '';
+    let requestInit: RequestInit = {};
+    requestInit.method = settings.method;
+    
+    if (settings.data) {
+      requestInit.body = <string>settings.data;
+    }
+    
+    //Calculate 'Accept' header value
+    let headers = settings.headers || {};
+    if (settings.dataType && settings.dataType.toLowerCase() === "json"){
+      headers['Accept'] = 'application/json, text/javascript, */*; q=0.01';
+    }else{
+      headers['Accept'] = '*/*';
+    }
+
+    //Set the request headers
+    requestInit.headers = {};
+    for (let i in headers) {
+      requestInit.headers[i] = <string>headers[i];
+    }
+    
+    return fetch(url, requestInit).then(response => {
+      return response.json();
```

Comment:
I am not sure if this is correct. If you have a dataType of `text` then calling `.json()` here on the response will fail.

Reply:
Now I'm calling `.json()` only if the requested dataType is `json` and `.text()` otherwise. We could mimic jQuery's behavior adding more parses for xml etc. but I'd rather keep things simple.